DFE-Digital/ecf-engage-and-learn
# Fix sign in bug

## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDEL-631

## Code review

### Is there anything weird that code reviewer should know?
The bug was caused by some lovely inner machinations of devise. I think `current_user` in some edge case was signing the user in - say, if there is no cached saved user and warden't `authenticate!` returned it a user.

So the solution is to not use warden's strategies for this stuff - I got pretty fed up trying to fix it without giving up on the strategy but it just was not very nice :/ 

But if you don't use any strategies, devise stops being nice - it doesn't draw the right routes, it doesn't redirect to sign-in when user is not authenticated on some pages, and so on... so I added a dummy strategy to keep it happy :( 

### Review Checks

## Product review

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Try signing in as `cip-less-ect@example.com`.
3. After you see error message, click on the service name.
4. You should not see the dashboard.
